src/main/java/org/apache/sysml/parser/DataExpression.java

```diff
@@ -246,31 +257,33 @@ else if (functionName.equals("matrix")){
 				if (unnamedParamCount == 5)
 					dataExpr.addMatrixExprParam(DataExpression.RAND_DIMNAMES,passedParamExprs.get(4).getExpr());
 				
-				if (unnamedParamCount > 5)
-					dataExpr.raiseValidateError("for matrix statement, at most 5 arguments supported (in order): data, rows, cols, byrow, dimname");
-								   
+				if (unnamedParamCount > 5) {
+					errorListener.validationError(blp, bcp, "for matrix statement, at most 5 arguments supported: data, rows, cols, byrow, dimname");
+					return null;
+				}
 				
 			} else {
 				// handle first parameter, which is data and may be unnamed
 				ParameterExpression firstParam = passedParamExprs.get(0);
 				if (firstParam.getName() != null && !firstParam.getName().equals(DataExpression.RAND_DATA)){
 					// throw exception -- must be filename as first parameter
```

Comment:
Hi Deron - should this 'throw exception' comment also be removed similar to change made in line 278?  Thanks, Glenn